diff --git a/node_modules/@web3-onboard/wagmi/dist/utils.js b/node_modules/@web3-onboard/wagmi/dist/utils.js
index 640c761..1bfd6d0 100644
--- a/node_modules/@web3-onboard/wagmi/dist/utils.js
+++ b/node_modules/@web3-onboard/wagmi/dist/utils.js
@@ -4,7 +4,7 @@ export const createWalletId = (walletLabel) => walletLabel.replace(/\s/g, '') +
 export async function createWagmiChains(chains, transports) {
     return (await Promise.all(chains.map(async (w3OChain) => {
         const { id } = w3OChain;
-        transports[fromHex(id, 'number')] = http();
+        transports[fromHex(id, 'number')] = http(w3OChain?.rpcUrl);
         return (await chainIdToViemImport(w3OChain));
     })));
 }
