diff --git a/node_modules/@web3-onboard/core/dist/index-8eb0493f.js b/node_modules/@web3-onboard/core/dist/index-8eb0493f.js
index 9d1bfdb..d65b3c1 100644
--- a/node_modules/@web3-onboard/core/dist/index-8eb0493f.js
+++ b/node_modules/@web3-onboard/core/dist/index-8eb0493f.js
@@ -2761,7 +2761,7 @@ async function getProvider(chain) {
         const { createPublicClient, http } = await import('viem');
         const publicProvider = createPublicClient({
             chain: viemChain,
-            transport: http()
+            transport: http(chain.rpcUrl)
         });
         viemProviders[chain.rpcUrl] = publicProvider;
     }
